src/Passes/Source/Commandline/ParameterParser.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    /// Parameter parser class which allows the developer to specify a set of default settings and
+    /// update those using the commandline argc and argv.
+    class ParameterParser
+    {
+      public:
+        using String      = std::string;
```

Comment:
This fragment is duplicated in a dozen of places. I would suggest to write it in a single place only, in one header, and including that header wherever needed.

Reply:
Done